moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -935,8 +941,11 @@ bool TrajectoryExecutionManager::distributeTrajectory(const moveit_msgs::RobotTr
 
 bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext& context) const
 {
-  if (allowed_start_tolerance_ == 0)  // skip validation on this magic number
+  // skip validation
+  if (not validate_trajectory_complete_) {
+    ROS_DEBUG_NAMED(name_, "Not validating if trajectory execution is complete");
```

Comment:
This validation concerns deviations of start start from current state. It's not related to completion!
This change should be reverted!